tools/gbcpal.c

```diff
@@ -0,0 +1,156 @@
+#define PROGRAM_NAME "gbcpal"
+#define USAGE_OPTS "[-h|--help] [-r|--reverse] out.gbcpal [in.gbcpal...]"
+
+#include "common.h"
+
+bool reverse;
+
+void parse_args(int argc, char *argv[]) {
+	struct option long_options[] = {
+		{"reverse", no_argument, 0, 'r'},
+		{"help", no_argument, 0, 'h'},
+		{0}
+	};
+	for (int opt; (opt = getopt_long(argc, argv, "rh", long_options)) != -1;) {
+		switch (opt) {
+		case 'r':
+			reverse = true;
+			break;
+		case 'h':
+			usage_exit(0);
+			break;
+		default:
+			usage_exit(1);
+		}
+	}
+}
+
+struct Color {
+	uint8_t r, g, b;
+};
+
+uint16_t pack_color(struct Color color) {
+	return (color.b << 10) | (color.g << 5) | color.r;
```

Comment:
True, but operator precedence beyond PEMDAS is obscure enough that people often use braces with bitwise operators anyway. (Especially because `|` and `&` precedence in C is a weird trap stemming from how they used to double as logical operators.) We do so in other tools, like png_dimensions.c's `(width_tiles << 4) | width_tiles`.